js/src/widget_ngl.ts

```diff
@@ -104,7 +104,10 @@ class NGLView extends widgets.DOMWidgetView{
 
     createStage(){
         // init NGL stage
-        var stage_params = this.model.get("_ngl_full_stage_parameters");
+        var stage_params = {
+            // Shallow copy so that _ngl_full_stage_parameters is not updated yet
+            ...this.model.get("_ngl_full_stage_parameters")
```

Comment:
Does the former code cause trouble?

Reply:
Yeah - see here: https://github.com/nglviewer/nglview/pull/1064#discussion_r1234069046

Basically stage_params was previously a reference to the object inside `_ngl_full_stage_parameters`, so when it was modified `_ngl_full_stage_parameters` was modified immediately. This meant that in `handleEmbed` it wasn't empty anymore, which would make checking it instead of `_ngl_is_initialized` more complicated and fragile. Making it a copy acts more like expected. The object isn't retrieved from the model anywhere except those two methods, and it's stored in the model already after `handleEmbed`, so I don't think this should otherwise change behavior.